### PR TITLE
add shimmer effect 

### PR DESCRIPTION
**Pull request by TalhaFaki**

### CODE REVIEW

**Comment by mertceyhan** [comment]
Hey @TalhaFaki, thank you so much for your interest in the project and for your contribution.  The shimmer is a good idea but could you check [this library](https://google.github.io/accompanist/placeholder/), please? It looks like a more easy solution.

**Comment by TalhaFaki** [comment]
Hi @mertceyhan , this lib looks better nice. I will check this day, thanks for feedback.

**Comment by TalhaFaki** [comment]
Hey @mertceyhan,  I replaced shimmer effect to accompanist shimmer. I hope, better good now.

File: buildSrc/src/main/java/com/mertceyhan/buildsrc/Dependencies.kt [hunk 1]
```diff
@@ -55,5 +55,6 @@ object Dependencies {
     const val roomKtx = "androidx.room:room-ktx:${Versions.roomVersion}"
     const val roomCompiler = "androidx.room:room-compiler:${Versions.roomVersion}"
     const val glance = "androidx.glance:glance-appwidget:${Versions.glance}"
+    const val accompanist = "com.google.accompanist:accompanist-placeholder:${Versions.accompanist}"
```
Comment (mertceyhan) [inline buildSrc/src/main/java/com/mertceyhan/buildsrc/Dependencies.kt:58]: I think `placeholder` would be a better name. `accompanist` is the name of the whole library but we use just place older, right?

`const val placeholder = "com.google.accompanist:accompanist-placeholder:${Versions.placeholderVersion}"`

File: buildSrc/src/main/java/com/mertceyhan/buildsrc/Versions.kt [hunk 2]
```diff
@@ -33,4 +33,5 @@ object Versions {
     const val accompanistSwipeRefresh = "0.21.0-beta"
     const val roomVersion = "2.3.0"
     const val glance = "1.0.0-alpha01"
+    const val accompanist = "0.22.0-rc"
```
Comment (mertceyhan) [inline buildSrc/src/main/java/com/mertceyhan/buildsrc/Versions.kt:36]: `placeholderVersion` would be better.

`const val placeholderVersion = "0.22.0-rc"`

File: app/src/main/java/com/mertceyhan/bitcoinmarket/features/market/ui/MarketScreen.kt [hunk 3]
```diff
@@ -115,6 +67,64 @@ fun MarketScreen(marketViewModel: MarketViewModel = hiltViewModel()) {
     }
 }
 
+@Composable
+fun ScreenContent(scrollState: ScrollState, viewState: MarketScreenViewState? = null, marketViewModel: MarketViewModel, isShowShimmer: Boolean) {
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/features/market/ui/MarketScreen.kt:71]: I'm not sure passing null `viewState` to UI 🤔 I think the UI shouldn’t care about the nullability. Do you have a better solution or thought for here?

File: app/src/main/java/com/mertceyhan/bitcoinmarket/core/ui/theme/Color.kt [hunk 4]
```diff
@@ -17,4 +17,9 @@ val gray900 = Color(0xFF23262D)
 val gray950 = Color(0xFFCCD0DD)
 val black = Color(0xFF000000)
 val blackTransparent20 = Color(0x07000000)
-val white = Color(0xFFFFFFFF)
\ No newline at end of file
+val white = Color(0xFFFFFFFF)
+val ShimmerColorShades = listOf(
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/core/ui/theme/Color.kt:21]: can we delete this if we don't need?

File: app/src/main/java/com/mertceyhan/bitcoinmarket/core/ui/ShimmerProperty.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package com.mertceyhan.bitcoinmarket.core.ui
+
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.placeholder.PlaceholderHighlight
+import com.google.accompanist.placeholder.shimmer
+
+object ShimmerProperty {
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/core/ui/ShimmerProperty.kt:10]: I think `PlaceholderProperty` would be better

File: app/src/main/java/com/mertceyhan/bitcoinmarket/components/Chart.kt [hunk 6]
```diff
@@ -41,6 +44,12 @@ fun Chart(
         modifier = modifier
             .fillMaxWidth()
             .requiredHeight(300.dp)
+            .placeholder(
+                visible = isShowShimmer,
+                color = ShimmerProperty.color,
+                shape = ShimmerProperty.shape,
+                highlight = ShimmerProperty.highlight
+            )
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/components/Chart.kt:52]: I think this is too much duplication, maybe we can create our own extension for reducing it.

```
package com.mertceyhan.bitcoinmarket.utils.extensions

import androidx.compose.ui.Modifier
import com.google.accompanist.placeholder.material.placeholder

fun Modifier.placeholder(visible: Boolean) = placeholder(
    visible = visible,
    color = PlaceHolderPromerty.color,
    shape = PlaceHolderPromerty.shape,
    highlight = PlaceHolderPromerty.highlight
)
```

File: app/src/main/java/com/mertceyhan/bitcoinmarket/components/Chart.kt [hunk 7]
```diff
@@ -18,6 +20,7 @@ import com.mertceyhan.bitcoinmarket.utils.extensions.setLineDataSet
 fun Chart(
     modifier: Modifier = Modifier,
     lineDataSet: LineDataSet? = null,
+    isShowShimmer: Boolean
```
Comment (mertceyhan) [inline app/src/main/java/com/mertceyhan/bitcoinmarket/components/Chart.kt:23]: I think `isPlaceHolderVisible` would be more meaningful cause there is no `shimmer` term in `Modifier`. If you agree, could you change all `isShowShimmer` naming to `isPlaceHolderVisible`

**Review by mertceyhan — CHANGES_REQUESTED** [review]

**Comment by mertceyhan** [comment]
Hey @TalhaFaki, I've put some comments on the PR, please mention your thoughts if you disagree. Also, the design looks a bit weird. The application follows CoinMarketCap's design, do you think we can do something similar? If you need help at any point, please ping me. Thanks a lot again, you rock 👏 

Screenshot_20220118-222513
Screenshot_20220118-222532

**Comment by TalhaFaki** [comment]
Hey @mertceyhan , Thank you very much for all your feedback, I'm closing pr. I'll open it back after a nice refactor.

**Pull request closed without merging**